src/main/java/com/suse/saltstack/netapi/client/impl/JDKConnection.java

```diff
@@ -70,6 +73,13 @@ private T request(String method, String data)
             connection.setDoInput(true);
             connection.setDoOutput(true);
 
+            // Timeouts may be specified on configuration
+            int connect_timeout = config.get(CONNECT_TIMEOUT);
+            if (connect_timeout >= 0) connection.setConnectTimeout(connect_timeout);
```

Comment:
Also, I am not sure if we want to trim negative here. If `config` is expecting to return 0, then it setting timeout to zero is same as disabling it, so you can just set it directly from getting it. But if configuration has negative value, this should be as configuration error, while this information is lost at this point.

But I don't have a strong opinion here. @renner, @moio ?